river/ensemble/adaptive_random_forest.py

```diff
@@ -0,0 +1,824 @@
+        self.max_depth = max_depth
+
+        if aggregation_method in self._VALID_AGGREGATION_METHOD:
+            self.aggregation_method = aggregation_method
+        else:
+            raise ValueError(f'Invalid aggregation_method: {aggregation_method}.\n'
+                             f'Valid values are: {self._VALID_AGGREGATION_METHOD}')
+
+    def predict_one(self, x: dict) -> base.typing.RegTarget:
+
+        if not self.models:
+            self._init_ensemble(features=list(x.keys()))
+            return 0.
+
+        y_pred = np.zeros(self.n_models)
+
+        if not self.disable_weighted_vote:
+            weights = np.zeros(self.n_models)
+            for idx, model in enumerate(self.models):
+                y_pred[idx] = model.predict_one(x)
+                weights[idx] = model.metric.get()
+
+                sum_weights = weights.sum()
+                if sum_weights != 0:
+                    # The higher the error, the worse is the tree
+                    weights = sum_weights - weights
+                    # Normalize weights to sum up to 1
+                    weights = weights / weights.sum()
+                    y_pred *= weights
+        else:
+            for idx, model in enumerate(self.models):
+                y_pred[idx] = model.predict_one(x)
+
+        if self.aggregation_method == self._MEAN:
+            y_pred = y_pred.mean()
+        else:
+            y_pred = np.median(y_pred)
+
+        return y_pred
+
+    def _get_base_model(self, seed: int):
+        return ARFHoeffdingTreeRegressor(
+            max_size=self.max_size,
+            memory_estimate_period=self.memory_estimate_period,
+            grace_period=self.grace_period,
+            split_confidence=self.split_confidence,
+            tie_threshold=self.tie_threshold,
+            binary_split=self.binary_split,
+            stop_mem_management=self.stop_mem_management,
+            remove_poor_attrs=self.remove_poor_attrs,
+            merit_preprune=self.merit_preprune,
+            leaf_prediction=self.leaf_prediction,
+            leaf_model=self.leaf_model,
+            model_selector_decay=self.model_selector_decay,
+            max_features=self.max_features,
+            nominal_attributes=self.nominal_attributes,
+            max_depth=self.max_depth,
+            seed=seed
+        )
+
+
+class BaseForestMember:
+    """Base forest member class.
+
+    This class represents a tree member of the forest. It includes a
+    base tree model, the background learner, drift detectors and performance
+    tracking parameters.
+
+    The main purpose of this class is to train the foreground model.
+    Optionally, it monitors drift detection. Depending on the configuration,
+    if drift is detected then the foreground model is reset or replaced by a
+    background model.
+
+    Parameters
+    ----------
+    index_original
+        Tree index within the ensemble.
+    base_model: ARFHoeffdingTreeClassifier
```

Comment:
Please, remove the type

Reply:
Done in 53c75bb9b45bfa0539d0c4bcd86bd3707603c5d9